bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,237 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.intesis.internal.handler;
+
+import static org.eclipse.smarthome.core.thing.Thing.*;
+import static org.openhab.binding.intesis.internal.IntesisBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.intesis.internal.IntesisConfiguration;
+import org.openhab.binding.intesis.internal.api.IntesisHomeHttpApi;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.AuthenticateData;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.dpval;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.info;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+
+/**
+ * The {@link IntesisHomeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+@NonNullByDefault
+public class IntesisHomeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisHomeHandler.class);
+
+    private static HttpClient httpClient = new HttpClient();
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private int refreshInterval = 30;
+    private String deviceIp = "";
+    private String password = "";
+    private String sessionId = "";
+
+    Gson gson = new Gson();
+
+    public IntesisHomeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("Start initializing!");
+        final IntesisConfiguration config = getConfigAs(IntesisConfiguration.class);
+
+        deviceIp = config.ipAddress;
+        password = config.password;
+        try {
+            httpClient.start();
```

Comment:
Better use `HttpClientFactory` and us `getCommonHttpClient` to get a httpclient. See other bindings on how it's used.

Reply:
Thanks @Hilbrand for pointing me into this direction. Will refactor my binding to address your comment.